binderhub/repoproviders.py

```diff
@@ -186,16 +187,32 @@ class GitRepoProvider(RepoProvider):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        url, resolved_ref = self.spec.rsplit('/', 1)
+        url, unresolved_ref = self.spec.rsplit('/', 1)
```

Comment:
While this is correct if we only accept SHA, it stops being correct if we need a branch name:

![image](https://user-images.githubusercontent.com/2069677/61045263-cf710780-a3da-11e9-8108-90b3b007457d.png)

Reply:
Good catch. There is no way to know which parts belong to the repository URL and which belong to the ref, so this is a fundamental issue with how the build URL is passed to the repo provider. There is actually no easy way to fix this, except for guessing by trying to split at each `/` and calling `git ls-remote`.

Reply:
I see other providers, e.g. `GithubRepoProvider` change what they mean with the ref. Could we try the same approach here?

Reply:
I just realized the namespec is url escaped, so using the regular `split` should work. Disregard my comment above.

Reply:
Fixed in 92ad34c